nextjsUIapp/src/app/knowledge-base/[slug]/[id]/page.tsx

```diff
@@ -1,122 +1,104 @@
 "use client";
 import SideBarKBFromConfig from "@/src/app/components/SideBarKBFromConfig";
-import {useEffect, useState} from "react";
+import { useEffect, useState } from "react";
 import enititycardmapperconfig from '@/src/app/components/enititycardmapper.yaml';
-import {getData} from "@/src/app/components/getData";
-import {fetchBoxes} from "@/src/app/components/fetchBoxes";
-interface EntityCard {
-    slug: string;
-    cardtype: string;
-    description: string;
-    name?: string;
-    sparql_query?: string;
-}
-const IndividualEntityPage = ({params}) => {
-    const {slug, id} = params;
+import { getData } from "@/src/app/components/getData";
+import {extractPredicateObjectPairs} from "@/src/app/components/helper";
+import {formatextractPredicateObjectPairs} from "@/src/app/components/helper";
+const IndividualEntityPage = ({ params }) => {
+    const { slug, id } = params;
     const [mainCardTitle, setMainCardTitle] = useState("");
     const [mainCardDescription, setMainCardDescription] = useState("");
-    const [extractedBoxes, setExtractedBoxes] =  useState<EntityCard[]>([]);//useState([]);
-    const [data, setData] = useState<any[]>([]);
+    const [extractedBoxes, setExtractedBoxes] = useState([]);
+    const [data, setData] = useState({});
     const [error, setError] = useState<string | null>(null);
-    const [loading, setLoading] = useState(false);
 
-    const baseurl = process.env.NEXT_PUBLIC_API_ADMIN_HOST || "http://3.134.90.242:8010";
-    const endpoint = process.env.NEXT_PUBLIC_API_QUERY_ENDPOINT || "query/sparql"; //default is "query/sparql"
-    const queryParameter = {sparql_query: decodeURIComponent(id)};
-    const fetchData = async () => {
+    const fetchData = async (queryParameter) => {
+        const baseurl = process.env.NEXT_PUBLIC_API_ADMIN_HOST;
+        const endpoint = process.env.NEXT_PUBLIC_API_QUERY_ENDPOINT || "query/sparql";
+
         try {
             const response = await getData(queryParameter, endpoint, baseurl);
-            if (response.status === 'success' && response.message?.results?.bindings) {
-                const bindings = response.message.results.bindings;
-                const vars = response.message.head.vars;
-                setData(bindings);
-            } else {
-
-                setError("Invalid data format");
+            if (response.status === "success" && response.message?.results?.bindings) {
```

Comment:
I'd probably default to evaluating the HTTP status code and not the `response.status`

Reply:
this is specific case where I get "success" from graphdatabse and wanted to check that. It's intentional